std/format.d

```diff
@@ -157,6 +157,11 @@ $(I Width):
     $(I empty)
     $(I Integer)
     $(B '*')
+$(I Underscore):
+    $(I empty)
+    $(B '_')
+    $(B '_') $(I Integer)
+    $(B '_*')
```

Comment:
This grammar change is incomplete. Where does `Underscore` fit in with the rest of the grammar?

And on that note, I see in your unittest examples that you inserted it between the width and the precision (i.e., `%5_4.5d`).  Personally, I don't like this. I think a better ordering would be to keep the width and precision together, and tack on the `_` afterwards, like this: `%5.5_4d`.